### PR TITLE
Upgrade site redirect settings to use the new design

### PR DESCRIPTION
**Pull request by olaseni**
#### Changes proposed in this Pull Request

This PR updates the settings for Site Redirects to use the new site design

#### Testing instructions

* Visit the domain settings page for a site redirect 
* Confirm that the new design is being used and looks like this:
Screenshot 2020-05-19 at 3 01 26 PM

### CODE REVIEW

File: client/my-sites/domains/domain-management/edit/navigation/index.jsx [hunk 1]
```diff
@@ -301,6 +302,19 @@ class DomainManagementNavigation extends React.Component {
...
 	}
 
+	getRedirectSettings() {
+		const { domain, selectedSite, translate } = this.props;
+
+		return (
+			<DomainManagementNavigationItem
+				path={ domainManagementRedirectSettings( selectedSite.slug, domain.name ) }
+				materialIcon="language"
+				text={ translate( 'Redirect settings' ) }
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/edit/navigation/index.jsx:312]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **34** times:
`translate( 'Redirect Settings' )` *ES Score: 9*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/use-the-new-design-for-site-redirects)

File: client/my-sites/domains/domain-management/edit/navigation/enhanced.jsx [hunk 2]
```diff
@@ -509,6 +510,19 @@ class DomainManagementNavigationEnhanced extends React.Component {
...
 	}
 
+	getRedirectSettings() {
+		const { domain, selectedSite, translate } = this.props;
+
+		return (
+			<DomainManagementNavigationItem
+				path={ domainManagementRedirectSettings( selectedSite.slug, domain.name ) }
+				materialIcon="language"
+				text={ translate( 'Redirect settings' ) }
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/edit/navigation/enhanced.jsx:520]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **34** times:
`translate( 'Redirect Settings' )` *ES Score: 9*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/use-the-new-design-for-site-redirects)

File: client/my-sites/domains/domain-management/edit/domain-types/site-redirect-type.jsx [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+				<WrapDomainStatusButtons className="domain-types__auto-renew-placeholder">
+					<p />
+				</WrapDomainStatusButtons>
+			);
+		}
+
+		return this.renderAutoRenewToggle();
+	}
+
+	planUpsellForNonPrimaryDomain() {
```
Comment (hambai) [inline client/my-sites/domains/domain-management/edit/domain-types/site-redirect-type.jsx:105]: I think we can remove this as site redirects should be outside the primary domains with plans only scope

File: client/my-sites/domains/domain-management/edit/domain-types/site-redirect-type.jsx [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+				</Card>
+				{ newDomainManagementNavigation ? (
+					<DomainManagementNavigationEnhanced
+						domain={ domain }
+						selectedSite={ this.props.selectedSite }
+						purchase={ purchase }
+						isLoadingPurchase={ isLoadingPurchase }
+					/>
+				) : (
+					<DomainManagementNavigation
```
Comment (hambai) [inline client/my-sites/domains/domain-management/edit/domain-types/site-redirect-type.jsx:178]: We'll soon clean up the old navigation component (`DomainManagementNavigation`) so no need to add it here at all. Just always render `DomainManagementNavigationEnhanced`

File: client/my-sites/domains/domain-management/edit/navigation/enhanced.jsx [hunk 5]
```diff
@@ -475,6 +476,8 @@ class DomainManagementNavigationEnhanced extends React.Component {
 			}
 		} else if ( domainTypes.MAPPED === domainType ) {
 			title = translate( 'Delete domain mapping' );
+		} else if ( domainTypes.SITE_REDIRECT === domainType ) {
+			title = translate( 'Delete site redirect' );
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/edit/navigation/enhanced.jsx:480]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **34** times:
`translate( 'Delete Site Redirect' )` *ES Score: 9*

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/3651469

Thank you @olaseni for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/3651469).

**Pull request merged**